backend/src/api.py

```diff
@@ -114,10 +100,27 @@ def register(
         side_effects: bool = False,
         deprecated: bool = False,
         default_nodes: List[DefaultNode] | None = None,
+        decorators: List[Callable] | None = None,
     ):
         def inner_wrapper(wrapped_func: T) -> T:
             p_inputs, group_layout = _process_inputs(inputs)
-            p_output = _process_outputs(outputs)
+            p_outputs = _process_outputs(outputs)
+
+            TYPE_CHECK_LEVEL = os.environ.get("TYPE_CHECK_LEVEL", "none")
+
+            if TYPE_CHECK_LEVEL != "none":
+                try:
+                    typeValidateSchema(wrapped_func, node_type, p_inputs, p_outputs)
+                except Exception as e:
+                    full_error_message = f"Error in {schema_id}: {e}"
+                    if TYPE_CHECK_LEVEL == "warn":
+                        logger.warning(full_error_message)
+                    else:
+                        raise TypeMismatchError(full_error_message) from e
```

Comment:
I think the `from e` is detrimental here. We don't care about the stack trace of any type errors, so this just makes the new error more complex without adding useful information.